src/main/scala/org/ergoplatform/modifiers/state/UtxoSnapshotManifest.scala

```diff
@@ -0,0 +1,94 @@
+package org.ergoplatform.modifiers.state
+
+import com.google.common.primitives.Ints
+import org.ergoplatform.modifiers.ErgoPersistentModifier
+import org.ergoplatform.modifiers.history.Header
+import org.ergoplatform.settings.{Algos, Constants}
+import scorex.core.ModifierTypeId
+import scorex.core.serialization.Serializer
+import scorex.core.validation.ModifierValidator
+import scorex.crypto.authds.{ADDigest, ADKey}
+import scorex.crypto.authds.avltree.batch.serialization.{BatchAVLProverManifest, BatchAVLProverSerializer, ProxyInternalNode}
+import scorex.crypto.authds.avltree.batch.{InternalProverNode, ProverNodes}
+import scorex.crypto.hash.Digest32
+import scorex.util.{ModifierId, bytesToId, idToBytes}
+
+import scala.util.Try
+
+case class UtxoSnapshotManifest(proverManifest: BatchAVLProverManifest[Digest32, Algos.HF],
+                                blockId: ModifierId,
+                                emissionBoxIdOpt: Option[ADKey])
```

Comment:
I don't see any checks for `emissionBoxIdOpt`, so malicious node may send incorrect `emissionBoxIdOpt` here and break our node. I propose to remove emissionBoxIdOpt from this class and search for it inside a state when it is recovered. Another option is to try to recover emission box from the first full block.

Reply:
@catena2w, fixed.